src/domain_bridge/generic_publisher.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
```

Comment:
Will these be factored out and made available to both the domain bridge and rosbag2? Seems like generally useful functionality.

Reply:
Yes. See https://github.com/ros2/rclcpp/pull/1452

Once that lands I'll remove the duplicate logic in this repo.

Reply:
Great.